File: PlusMinus.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {
    public static void plusMinus(List<Integer> arr) {
        int positiveCount = 0;
        int negativeCount = 0;
        int zeroCount = 0;
        int n = arr.size();
        
        for (int i = 0; i < n; i++)
        {
            int num = arr.get(i);
            if (num < 0)
            {
                negativeCount++;
            }
            else if (num == 0)
            {
                zeroCount++;
            }
            else
            {
                positiveCount++;
            }
        }
        
        float positiveRatio = positiveCount / (float) n;
        float negativeRatio = negativeCount / (float) n;
        float zeroRatio = zeroCount / (float) n;
        
        System.out.printf("%.6f%n", positiveRatio);
        System.out.printf("%.6f%n", negativeRatio);
        System.out.printf("%.6f%n", zeroRatio);
    }
}

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(bufferedReader.readLine().trim());

        List<Integer> arr = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
            .map(Integer::parseInt)
            .collect(toList());

        Result.plusMinus(arr);

        bufferedReader.close();
    }
}

File: PlusMinus.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'plusMinus' function below.
#
# The function accepts INTEGER_ARRAY arr as parameter.
#

def plusMinus(arr):
    # Initialize counters for positives, negatives, and zeros
    positiveCount = 0
    negativeCount = 0
    zeroCount = 0
    
    # Get the total number of elements in the array
    n = len(arr)
    
    # Loop through the array and update counters
    for num in arr:
        if num > 0:
            positiveCount += 1
        elif num < 0:
            negativeCount += 1
        else:
            zeroCount += 1
    
    # Calculate the ratios
    positiveRatio = positiveCount / n
    negativeRatio = negativeCount / n
    zeroRatio = zeroCount / n
    
    # Print the results with 6 decimal places
    print(f"{positiveRatio:.6f}")
    print(f"{negativeRatio:.6f}")
    print(f"{zeroRatio:.6f}")

if __name__ == '__main__':
    n = int(input().strip())

    arr = list(map(int, input().rstrip().split()))

    plusMinus(arr)
